champions/src/main/java/me/mykindos/betterpvp/champions/weapons/weapons/ThrowingWeb.java

```diff
@@ -85,7 +90,10 @@ public void activate(Player player) {
     @EventHandler
     public void onInteract(PlayerInteractEvent event) {
         if (event.getAction().isLeftClick() && matches(event.getItem())) {
-            activate(event.getPlayer()); // also activate on left click
+            String name = PlainTextComponentSerializer.plainText().serialize(getName());
```

Comment:
You should probably use `BPVPItem#getKey` instead of `BPVPItem#getName` since the former one has to be unique for all items and is already a string, whereas the latter one is an adventure component, and for classes that use the default constructor is the same component with content `Unintialized Weapon`, meaning cooldown would be shared across those weapons

Reply:
This is using the same thing as where the normal cooldown is being applied. I think it makes more sense to refactor both (it makes more sense). But getName would be overwritten when the item is loaded, so not an issue there.

Reply:
Still should probably use the key anyway, better practice than to use the display name

Reply:
the display name is also what is used to display to the player, which key was not really intended to

Reply:
Oh I see what you mean, the cooldown manager displays that string. All good then